src/components/main/sidebar/chat/mod.rs

```diff
@@ -229,7 +229,10 @@ pub fn Chat<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                         hr{}
                         ContextItem {
                             icon: Shape::XMark,
-                            onpressed: move |_| {},
+                            onpressed: move |_| {
+                                let state = use_atom_ref(&cx, STATE).clone();
```

Comment:
this is very bad. it violates the [rules of hooks](https://dioxuslabs.com/guide/interactivity/hooks.html). hooks like `use_state`, `use_ref`, and `use_atom_ref` should be defined at the top of the function, outside of any other scope, such as conditionals, loops, or closures.

If you ever need to use the same hook within two closures, define a separate variable for each closure: `let state2 = state.clone()` and then use `state` in the first closure and `state2` in the second closure.